jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java

```diff
@@ -167,7 +167,7 @@ public String getHostHeader()
 
     public String getViaHost()
     {
-        return _viaHost;
+        return _viaHost == null ? viaHost() : _viaHost;
```

Comment:
Don't really want to compute `viaHost()` all the times I call this getter.
I would leave it as it was originally.

Reply:
we should assign the result to the field to avoid this. As discussed on the mailing list, we're not using this class as a Servlet, so the `_viaHost` value was null. I wasn't gonna mess with that, but it also came up in the tests where the host was null. I didn't want to expect `HTTP/1.1 null`, so I did this. I'll fix it to save the results in the field though, so it's only called once.